### ISSUE TITLE
Spatial types support

### ISSUE THREAD

**Issue by teebot** [prisma/prisma2#1798]
I was thinking how that feature could work but it’s probably a bit naive given I don’t know the implementation details of prisma2 :

- Have a schema annotation @@spatial so that migrate would create a geometry type in postGIS, a geojson field in mongo etc.  
- **generate** could type this to a geoJSON definition
- **client** would translate queries to St_FromGeoJSON in postGIS, leave the geoJSON as is in Mongo 

The rationale behind this is that you basically want **migrate** to create the right spatial type in your db, currently schema/code-first does not work and introspect detects spatial as a string. Yuo also want **client** to perform CRUD on tables containing spatial types, currently CRUD fails if the spatial type is in the select so a `.raw` request is the only workaround.

**Comment by zifahm** [comment]
is there a workaround for this?

**Comment by teebot** [comment]
Yes!
- When you introspect geometry types will be marked as strings in your schema
- Exclude these types from your crud queries
- when you need them make a raw query and expose as geojson then json parse the string

**Comment by zifahm** [comment]
I would love to see and example @teebot :)

**Comment by tctrautman** [comment]
Hi all.  First time poster here, coming to Prisma from the Redwood community.  Thanks for the work you all are doing -- I've really enjoyed working with Prisma 2 and Prisma migrate so far.

Is there a place I can check to see where geo support falls on the Prisma roadmap or, if not, is that something you could speak to here? (Apologies if I missed this in my initial search.)

**Comment by jaywalklabs** [comment]
@tctrautman 
cant speak on actual support in the schema annotation but the real issue is the lack of custom types (i.e geography)
hopefully its covered by this https://github.com/prisma/prisma/issues/446 
which is marked in the next milestone (2.4.0) https://github.com/prisma/prisma/milestone/41

once thats in im hoping that you dont have to fight against migrate that its trying to remove columns thats not in the schema etc. running raw queries is really not an issue for me personally

**Comment by tctrautman** [comment]
Thanks @jaywalklabs -- this is really helpful.

**Comment by thebiglabasky** [comment]
Hi,
Supporting GIS types is not currently planned. We have indeed started picking up native types as mentioned above here #446 but we first focus on the most common types which don't require specific client-side mapping yet.
The roadmap is available at https://pris.ly/roadmap if you want to have an overview of the topics we're tackling and ideas about our timelines

**Comment by bkiac** [comment]
@thebiglabasky My only real issue with unsupported GIS types is being unable to use the migration tool. Are you planning to / is it feasible to implement a type or function to flag a field as any type so the migration tool can ignore it?

**Comment by thebiglabasky** [comment]
@bkiac Interesting, that's good context, thanks.
I don't believe we have ironed out the details at this point, but I will surface that idea to the team.

**Comment by jebenois** [comment]
+1

**Comment by theswerd** [comment]
+1

**Comment by gilneto8** [comment]
Will this be rolling out soon?

**Comment by Jolg42** [comment]
You can now try `2.16.0`

`@@ignore` & `Unsupported` makes it possible to use PostGIS for example, as `$raw` queries, and Migrate will not drop these columns.

Release Notes:
https://github.com/prisma/prisma/releases/tag/2.16.0

Also check:
https://github.com/prisma/prisma/issues/4531#issuecomment-771893085

Daniel created a PostGIS example using `$raw` queries with the client and using latest Migrate Preview features:
https://github.com/2color/interest-points/blob/master/prisma/schema.prisma

Truncated example:
```
model berlin_pois {
  gid    Int                      @id @default(autoincrement())
  osm_id String?                  @db.VarChar(10)
  code   Int?                     @db.SmallInt
  fclass String?                  @db.VarChar(28)
  name   String?                  @db.VarChar(100)
  geom   Unsupported("geometry")?
  @@index([geom], name: "berlin_poi_idx")
  @@map("berlin-pois")
  @@ignore
}

model spatial_ref_sys {
  srid      Int     @id
  auth_name String? @db.VarChar(256)
  auth_srid Int?
  srtext    String? @db.VarChar(2048)
  proj4text String? @db.VarChar(2048)
  @@ignore
}
```

We are looking for feedback and are planning to release Prisma Migrate out of Preview soon 😄

**Comment by benjamintd** [comment]
Hi! Thanks for the example using the unsupported types. 

Is it possible to use a middleware to handle these unsupported types?

A common use case would be: 
- on create/update, all geometry types can be parametrized as geojson the middleware would convert them  using `
ST_GeomFromGeoJSON`
- on read, all geometry types are read as geojson using `
ST_AsGeoJSON`. 

Can this be done with the current middleware API, or do we have to resort to raw queries ?

**Comment by Jolg42** [comment]
Hi @benjamintd 

So you'll need to use raw queries since `Unsupported` fields are not surfaced in the client at all.

**Comment by baba43** [comment]
I have a `POINT` field in my MySQL db that gets annotated by `Unsupported("point")`.

Is there any simple way or workaround that would allow prisma to at least fetch this field in regular queries?

**Comment by Jolg42** [comment]
@baba43 you'll need to use the $raw queries, someone opened a related feature request https://github.com/prisma/prisma/issues/5683

**Comment by baba43** [comment]
Well, when annotating the field with `Unsupported("point")?` I get this error for rows where this field is `NULL`:

```
PrismaClientKnownRequestError2 [PrismaClientKnownRequestError]:
Invalid `prisma.queryRaw()` invocation:
Raw query failed. Code: `N/A`. Message: `N/A`
  code: 'P2010',
  clientVersion: '2.18.0',
  meta: { code: 'N/A', message: 'N/A' }

```
The error disappears as soon as I remove the field from my schema completly. 
However, then the field is then added again and again with each `prisma introspect`.

**Comment by janpio** [comment]
Please open a new issue about this @baba43 and we are happy to look into this and possibly fix it if we can reproduce.

**Comment by VinayaSathyanarayana** [comment]
Any idea on when this will be available natively?

**Comment by garretteklof** [comment]
Curious about the progress on this as well. The project I'm leveraging this on is geospatial heavy and looking forward to the day I can drop all the `$raw` querying - the migrations don't always play well with geospatial types either (have to reset often because 'type Geometry not recognized', etc).

**Comment by tomhoule** [comment]
Hi @garretteklof ! We should have full support in migrate and introspection already. We would be super grateful for an issue explaining your problems, in case you are encountering something that looks like a bug. Could it be that a `CREATE EXTENSION 'postgis'` is missing from your migrations? It does need to be there for migrate to work (because the extension needs to be enabled on the shadow database as well — it's something we need to document better).

**Comment by thebiglabasky** [comment]
We also are tracking this through the PostGIS support issue here: https://github.com/prisma/prisma/issues/2789

This is something we definitely want to get to, but have been pushing a bit so far. I'll let @matthewmueller comment once the team will have capacity (currently working hard on MongoDB support) but assume it to happen within 2021.

**Comment by garretteklof** [comment]
@tomhoule okay next time it occurs I'll make an attempt to isolate the issue. I definitely have`CREATE EXTENSION postgis;` along with other initializations with the initial migration. Because of the geospatial aspect, I often run the migrations as `--create-only` first because I need to update various columns  `SRID` etc.

**Comment by garretteklof** [comment]
@matthewmueller out of morbid curiosity, is this still on the docket to begin in 2021?

**Comment by aeddie-zapidhire** [comment]
It's worth noting that using `Unsupported("point")` will cause the write methods on the `Delegate` class to be omitted, which is somewhat of a pain. I've tried using `location String` in the schema, but using `POINT` in the SQL, but there is an object conflict and postgres throws an error.

So what I am going to try is (in case it is food-for-thought for anyone else):

1. Use `location String` in `prisma.schema`
2. Leave `"location" TEXT NOT NULL,` in the migration and add `"_location" POINT,`
3. After any write operation run a raw sql to copy the `location` to `_location`, casting to a point. (or, I could add a trigger to do the same thing).
4. Do my searches using raw SQL based on the `_location` column.

**Comment by garretteklof** [comment]
Please, for the love of all that is good and holy, get this feature in.

**Comment by masihjahangiri** [comment]
May you attention to our request, please.

**Comment by oyatek** [comment]
we need this too!

**Comment by MurzNN** [comment]
> For what it's worth I ended up going with separate `lat` and `long` fields in the end. Reason for this was that the `ll_to_earth` function takes those 2 arguments so the case to try and make a `point` column type work evaporated.

Do you store them as decimal numbers, or as string? And is there any technique available to filter points by coordinates+radius, using that storage type?

Also for storing simple GPS points - there is a separate issue https://github.com/prisma/prisma/issues/5984 - so maybe implementing that feature will be the good first step forward to this issue?

**Comment by aeddie-zapidhire** [comment]
@MurzNN 

Stored as decimals. Migration file will look something like this:

```sql
CREATE EXTENSION IF NOT EXISTS "cube";
CREATE EXTENSION IF NOT EXISTS "earthdistance";

-- ...

CREATE TABLE "thing" (
-- ...
    "latitude" DOUBLE PRECISION NOT NULL,
    "longitude" DOUBLE PRECISION NOT NULL,
-- ...
);
```

Then in my query I have to do some funky stuff.

```js
const FIX = 0.9999999999

// ...

// Getting the lat and long from the REST query params
const { lat, long } = params

// Running a raw query
const results = await this.db.$queryRaw<any[]>`
SELECT *, earth_distance(
  ll_to_earth(${+lat * FIX}, ${+long * FIX}),
  ll_to_earth(s.latitude, s.longitude)
) AS distance
FROM thing
ORDER BY distance ASC
`
```

So the `lat` and `long` are coming from the API query parameters, for example `GET /things?lat=1&long=1`.

I had to add the `FIX` because there seems to be a bit of a bug in Prisma when is sees a whole number. It doesn't encode it properly so I had to fudge it so that the template literal was always a decimal. It's a dirty workaround but couldn't work around it any other way.

Not sure if that helps your use case, but that will give me, for example, a list of "things" in order of proximity to my current location.

**Comment by lucysutton91** [comment]
+1

**Comment by freddydumont** [comment]
@tmcw Sorry to ping you but I'm wondering what you schema looks like to use a geometry column with Prisma? I understand you still need to write raw SQL queries. I don't quite understand what the Gist index brings though.

**Comment by freddydumont** [comment]
Thanks @tmcw that's really helpful.

**Comment by floelhoeffel** [comment]
Hello everyone 👋

We are kicking off design of GIS support for Prisma!

If you would like to help, please let us learn about your GIS needs!

Via survey:
Please fill out our [GIS User Research Survey](https://prisma103696.typeform.com/to/p8xo5o95)

Via email:
Alternatively, just shoot us an an email to [feedback@prisma.io](mailto:feedback@prisma.io).
The questions we have are (all in the context of GIS):

- what is your GIS use case?
- what database do you use?
- what column types do you use?
- what queries do you use?
- what indexes do you use?
- what kind of defaults do you use?

Thank you!

**Comment by DanielSmith** [comment]
It's September 2022.  I wanted to try Prisma, but not being able to search MongoDB by proximity means that I will stick with Mongoose for now.   A snippet from my Node.js, where I am building up a query (and note: this was working fine in Mongoose in 2017):

```js
  retVal = {
    "$near": {
      "$geometry": {
        "type": "Point",
        "coordinates": [ longitudeFloat, latitudeFloat ]                                
      },
      "$maxDistance": 2000
    }
  };
```
Mapping apps often need to search by proximity. Prisma needs to support this.

**Comment by murbanowicz** [comment]
@floelhoeffel Hi! Do you have any update on this? I am right on the edge of choosing an ORM for my next project where I need a lot of GIS stuff. Could you share anything?

**Comment by AdsonCicilioti** [comment]
Is there a plan to map it to a specific Release?

Working with GeoLocation data has been increasingly common and powerful nowadays.

**Comment by floelhoeffel** [comment]
@murbanowicz we are wrapping up discovery as we speak. We hope to pickup implementation in Q4 but can only tell after planning (which happens early December). Thank you for your patience 🙏

**Comment by DanielSmith** [comment]
It's amazing that this is taking so long (over 2.5 years...)  Prisma is not someone's hobby project - it's a company with over $50M in financing.  And..and.. this is very much a solved problem (Mongoose for > 5 years now, and others).  Kick it up a notch.

**Comment by Oreilles** [comment]
@floelhoeffel I'm planning to use Prisma for my current work projects, and support for geospatial data types will be a requirement, I'd love to get involved in the implementation process - I think I have a pretty good knowledge of the issues you might encounter, as my previous work at [Directus](https://github.com/directus/directus) specifically involved building an GeoJSON IO layer compatible with Postgres, MySQL, Oracle, SQLite and SQL Server. Please get in touch if you're interested in getting me involved.

**Comment by janpio** [comment]
That sounds wonderful. How can we contact you? You are hard to google 😆 Feel free to send an email to jan@prisma.io and I will forward to @floelhoeffel.

**Comment by Oreilles** [comment]
Thanks Jan, I sent you an email 👍

**Comment by dylel** [comment]
I see that on the roadmap this has moved from work in progress back to planned (after being WIP for about 2months). Does this mean that we won't be getting this anytime soon? I think a lot of us were hopeful that this would in the December or in the January release. Can we get a rough eta, like will this be soon or months?

**Comment by janpio** [comment]
If you define "soon" as "less than a month or two months", then yes we will not get this done in that timeframe. With the complexity we uncovered, we unfortunately needed to revisit our prioritization and reflected that in our roadmap. We still want to and will build this - but not in the timeframe we initially hoped.

**Comment by MurzNN** [comment]
@janpio Could you please consider the implementation of `Point` data type (described in https://github.com/prisma/prisma/issues/5984) as a first step? Seems it is not so hard to implement, like full Spatial features.

**Comment by janpio** [comment]
Leave a comment there in the issue how you would imagine this would work, what new types and API methods would be useful or required for that standalone. Thanks.

**Comment by Oreilles** [comment]
Hi @janpio, I didn't get an answer to the email I sent. Did you (or me) not receive it, or are you not interested anymore in collaborating on this specific issue ?

**Comment by dylel** [comment]
@janpio I appreciate the update, as you still seem to be in the api design phase I would think looking at something like entity framework from c# as they have the best implementation/support i've seen for spatial queries and operations that I know of

**Comment by LeulAria** [comment]
Do we have support for this yet or should i use another ORM

**Comment by DanielSmith** [comment]
> Do we have support for this yet or should i use another ORM

The issue will be Three Years old in another week + they just had some layoffs January 20th.  My personal choice would be to look elsewhere.  This is very much a solved problem, and coming from projects that have not had the benefit of $50,000,000 in financing.

**Comment by okonon** [comment]
We are diving in to geo world too lol hopefully i will be able to test/provide feedback once i get in to the weeds.

**Comment by altarrok** [comment]
Is this feature rolled back to planning? Why?

**Comment by altarrok** [comment]
We are also in need of this feature

**Comment by luis-cicada** [comment]
We need also this functionality 🚀

**Comment by vinaykharayat** [comment]
I need this feature as I have to manually create geospatial query every time I push my schema and it removes my index in mongodb and I have to recreate it.

**Comment by okonon** [comment]
@VinayaSathyanarayana hmm i thought there was an example with one of the plugins `typescript/postgis-express` https://github.com/prisma/prisma-examples/tree/c69103dcf06878e7df19b85c91004066cf54fcbe/typescript/postgis-express 

Hope this helps

**Comment by vinaykharayat** [comment]
> @VinayaSathyanarayana hmm i thought there was an example with one of the plugins `typescript/postgis-express` https://github.com/prisma/prisma-examples/tree/c69103dcf06878e7df19b85c91004066cf54fcbe/typescript/postgis-express
> 
> Hope this helps

If this is for me (I think you tagged wrong person). Actually I am using mongo DB

**Comment by Velozity** [comment]
I had a dream that spatial types support was added - reality is cruel

**Comment by Reevo55** [comment]
It is hilarious that "next-gen ORM" with 1 394 472 downloads per week doesn't have one of the most basic database features after three years after the issue release

**Comment by DanielSmith** [comment]
Yep, most of those downloads are automated, but .. the point still stands.  To be clear, this is a company with > 50M in financing.. it is not someone's hobby project.  It is not unreasonable to expect support for something as basic as mapping.  This issue is > 3 years old. .and . .and. .it is very much a solved problem (Mongoose and TypeORM, to name two).  It's ridiculous.  (taking myself off of watching this issue, I just dont "believe" it's going to be addressed)

**Comment by LeulAria** [comment]
> i use another ORM

it's jus hype >50M revenu and no one to pick a backlog task...what a company maybe hire me guy's lol..

**Comment by patryk-smc** [comment]
With new prisma extensions, creating your own GIS methods is trivial. Maybe it's not native but definitely more handy than using raw queries

**Comment by theronic** [comment]
How do I specify a custom column type? Asked on SO: https://stackoverflow.com/q/76920525/198927

**Comment by marisancans** [comment]
Any updates on this? Rewrote all the schemas for my project just to see that GeoJson is not supported in prisma for mongodb. What a waste of time

**Comment by theronic** [comment]
@marisancans did the same. Ended up ditching Prisma for everything except Migrations. I also could not get `prisma.$executeRaw` to template GEOMETRY inputs correctly. Just write SQL.

IMO, avoid ORMs. They don't help you. Query composition is good. Migrations are good. But ORM has only ever made my life harder.

**Comment by Tjklimoski** [comment]
Just voicing my support for this topic, as the ability to specify a index type while using `@@index()` with mongodb would be greatly appreciated. There's type support for already for PostgreSQL and CockroachDB so I'm hoping one for MongoDB is in the works, as having GeoSpatial index support would be fantastic.

**Comment by shadrech** [comment]
This should really be made into a priority. I chose prisma for a new project having worked with Typeorm in last project, I'm now regretting the switch as geospatial features are now very much integral to my current project. Using prisma with postgis, when I assign the geography column via `Unsupported("geography")`, all queries thereafter do not acknowledge that field at all. Which means I have to rewrite all my queries using `$rawQuery` which isn't ideal. Please try and work on supporting geospatial data
Any update on the PR https://github.com/prisma/prisma-engines/pull/4208?

**Comment by roland-pep** [comment]
big bummer

**Comment by isaacrowntree** [comment]
Still not working 2023?

**Comment by kakasoo** [comment]
I need this feature.

**Comment by okonon** [comment]
I am at the point in my project to start working on spatial stuff and bummed that PostGIS is not integrated easily with Prisma. looked at prisma-engines PR and see if i can test it out

**Comment by fcristel** [comment]
So.....still not available? This is very basic functionality in our days. Please add it to roadmap

**Comment by trishanduong** [comment]
Been loving using Prisma so far, but I'm super sad to see this geo/spatial data isn't supported yet :( Hope to see this get implemented soon!

**Comment by elbassel** [comment]
How many years will be enough to get this out?

**Comment by MangMax** [comment]
Recently started to use the prisma, try to query in mysql found that actually still does not support, I'm very eager to know the progress of this feature, this is a mature orm is not a luxury!
```
`SELECT id FROM land_info WHERE ST_Contains(geom, ST_PointFromText('POINT(${input[0]} ${input[1]})'))`
```

**Comment by walterjansma** [comment]
Any update on the timeline for the PostGIS support? I see it's currently 'planned': https://prismaio.notion.site/Prisma-ORM-Roadmap-50766227b779464ab98899accb98295f

**Comment by roland-pep** [comment]
> Any update on the timeline for the PostGIS support? I see it's currently 'planned': https://prismaio.notion.site/Prisma-ORM-Roadmap-50766227b779464ab98899accb98295f

it's been planned since the beginning of time.

**Comment by konarskis** [comment]
Hi, what are the chances we can have it in prod within the next month or so? It would be really crucial....

**Comment by konarskis** [comment]
> Zero. We are not actively working on this right now, and it will not be possible get this our and production ready in a month (or two). Sorry.

Got it, thanks for a prompt reply!

**Comment by jakobsuckow** [comment]
Is there any foreseeable timeline when this will be worked on, if ever ?

**Comment by janpio** [comment]
Yes, we are pretty confident we will work on this _some time_. We do not communicate ETAs or concrete timelines though. But it is definitely higher on the list than some other issues which we will definitely not touch this quarter or year.

**Comment by LeulAria** [comment]
i guess time to switch orm

**Comment by ghyath5** [comment]
still waiting for this feature
looking forward

**Comment by mtjosue** [comment]
At this point I would be happy with a simple "official" "workaround" by prisma, thanks to all who have posted your bits of knowledge.

**Comment by KrustyC** [comment]
any update or ETA on this?

**Comment by marcodali** [comment]
I don't want to use raw queries that's why I'm using prisma in the first place to not deal again with SQL. What we want is an easy way to use the awesome Point datatype that Postgres offer, just simply as that.

Can you make it possible or not?

If not please at least have the decency of saying NO we can't, it's not in our plans, but this kind of request have been open for around 3 years and still open which talks very bad about you as development team.

No hard feelings, just saying the truth. I hope this message does not offend to anyone.

**Comment by LeulAria** [comment]
they have like 3k issues, this issues it's dark web here hhhh :)

**Comment by janpio** [comment]
@marcodali 6 comments above I shared our (Prisma's) perspective: https://github.com/prisma/prisma/issues/1798#issuecomment-1960082047

So yes it is possible, we will implement this one day, no there is no ETA for it. And no, it is not as simple as just supporting `Point` as a datatype because then you would still not be able to run any queries that use this datatype.

I know this is frustrating to some of you, it is for me as well. But we are a small team working on a massively used library that has many, many reported bugs and missing features - and we can only do a limited number of things at a time. Hopefully soon this will be GIS support.

**Comment by mtjosue** [comment]
The 10% of repositories using geometry types is more like 40 million+. 
I wouldn't consider that number as "some of you"

There seems to be "some" lack of understanding of the power and need for geometry types in the future. 

These types enable a plethora of functionalities that have always been transformative in how we interact as humans. 

Consider the major apps that use geometry types like Google Maps, Airbnb, and even more recreational apps like 
Pokémon Go, which by the way, has generated over $5 billion in its lifetime, 566 Million in revenue in 2023, Prisma. 

I understand it might not be a big money-maker for you all initially, but the impact is not just to "some of us" and it's disappointing to be viewed this way and not be given a straight answer.

**Comment by LeulAria** [comment]
> Drizzle just made it available. Just saying...
> https://github.com/drizzle-team/drizzle-orm/releases/tag/0.31.0

Oh nice

**Comment by konarskis** [comment]
> Drizzle just made it available. Just saying... https://github.com/drizzle-team/drizzle-orm/releases/tag/0.31.0

Whoop whoop

**Comment by acidjazz** [comment]
requested in 2020.. still no support.

**Comment by ananidze** [comment]
Any updates on the progress of geospatial query support? 👀

**Comment by ananidze** [comment]
Any updates on the progress of geolocation query support?

**Comment by colelloa** [comment]
+1 on this issue, please!

**Comment by fcanmekikoglu** [comment]
+1

**Comment by anaibol** [comment]
+1

**Comment by Oreilles** [comment]
To people commenting "+1" repeatedly: this is accomplishing nothing, except annoying both the maintainers and the people who subscribed to this issue in order to get actual updates in due time. Just stop.

**Comment by acidjazz** [comment]
> To people commenting "+1" or asking for updates repeatedly: this is accomplishing nothing, except annoying both the maintainers and the people who subscribed to this issue in order to get actual updates in due time. Just stop.

This is a common pattern for github issues, and this feature is extremely important to all of us. 

You're more than welcome to unsubscribe as I'm sure some maintainers have already, there is a button right at the top to do so.

**Comment by anaibol** [comment]
+1

**Comment by SarenT** [comment]
-1
(it is annoying to get spam e-mail with bunch of +1s, while following this issue for substantial details). You can set a thumbs up and follow the topic or if you are so eager, implement yourself and send a pull request.

**Comment by millette** [comment]
Might I suggest locking issues after 4 years? Admins/contributors can keep us updated.

**Comment by acidjazz** [comment]
> -1 (it is annoying to get spam e-mail with bunch of +1s, while following this issue for substantial details). You can set a thumbs up and follow the topic or if you are so eager, implement yourself and send a pull request.

If only there was a way to Unsubscribe

<img width="162" alt="image" src="https://github.com/user-attachments/assets/d2bc7419-fb2a-437f-841c-c4594f9f001d">

**Comment by SarenT** [comment]
> If only there was a way to Unsubscribe

If only people could understand what they read:
> **_while following this issue for substantial details_**

**Comment by DanielSmith** [comment]
> Drizzle just made it available. Just saying... https://github.com/drizzle-team/drizzle-orm/releases/tag/0.31.0

Great and props to Drizzle for doing this.. for free!  Prisma has received over $40M in financing, yet still have not managed to address this issue for over 4 1/2 years.

**Comment by Sn-Kinos** [comment]
> > To people commenting "+1" or asking for updates repeatedly: this is accomplishing nothing, except annoying both the maintainers and the people who subscribed to this issue in order to get actual updates in due time. Just stop.
> 
> This is a common pattern for github issues, and this feature is extremely important to all of us.
> 
> You're more than welcome to unsubscribe as I'm sure some maintainers have already, there is a button right at the top to do so.

> > -1 (it is annoying to get spam e-mail with bunch of +1s, while following this issue for substantial details). You can set a thumbs up and follow the topic or if you are so eager, implement yourself and send a pull request.
> 
> If only there was a way to Unsubscribe
> 
> <img alt="image" width="162" src="https://private-user-images.githubusercontent.com/967369/372170573-d2bc7419-fb2a-437f-841c-c4594f9f001d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE5Mjk2NTYsIm5iZiI6MTczMTkyOTM1NiwicGF0aCI6Ii85NjczNjkvMzcyMTcwNTczLWQyYmM3NDE5LWZiMmEtNDM3Zi04NDFjLWM0NTk0ZjlmMDAxZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTExOFQxMTI5MTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00ZmQ3MzFjOWY1MzEyOGNhZWI5MTkxZDE4MzdhNDJhMDUzODNhYmM3OWM1ZDk2MWMyZTNiZWJmYjY2NGM4ODE1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.WUdAco8zucn04K5nkAQzjtZMD7M6EVoUHe_oljCtvVI">

He is very intelligently discouraging admins from paying attention to the issue. If more admins unsubscribe from issues in this way, they won't even get notified of important comments. I think that's a pretty terrible. Of course, the +1'ers are the most problematic, but I think it's a bad idea to encourage them in this way.

**Comment by LeulAria** [comment]
This thread is still open 😱

**Comment by anaibol** [comment]
+1

**Comment by acidjazz** [comment]
+1, excited to see more progress on the PR!

**Comment by mc-luisg** [comment]
+1

**Comment by Em1lT** [comment]
+1

**Comment by mateoghione99** [comment]
+11111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111

**Comment by AntonyBaasan** [comment]
+1

**Comment by bnhovde** [comment]
+1

**Comment by altarrok** [comment]
> To people commenting "+1" or asking for updates repeatedly: this is accomplishing nothing, except annoying both the maintainers and the people who subscribed to this issue in order to get actual updates in due time. Just stop.

by initiating a discussion unrelated to this issue, on this issue, you are contributing to the spam. As more than 5+ comments have been made about this discussion. Please don't respond and accept that +1 ers will +1, and feel free to open a new issue about this. +1 spams are at least easy to disregard, when you open a discussion it is not disregarded until some thought has been given to it. You are pouring oil to the flame..

**Comment by SarenT** [comment]
> > To people commenting "+1" or asking for updates repeatedly: this is accomplishing nothing, except annoying both the maintainers and the people who subscribed to this issue in order to get actual updates in due time. Just stop.
> 
> by initiating a discussion unrelated to this issue, on this issue, you are contributing to the spam. As 5+ comments have been made about this discussion. Please don't respond and accept that +1 ers will +1, and feel free to open a new issue about this. +1 spams are at least easy to disregard, when you open a discussion it is not disregarded until some thought has been given to it. You are pouring oil to the flame..

by initiating a discussion unrelated to this issue, on this issue, you are contributing to the spam. As 6+ comments have been made about this discussion. Please don't respond and accept that people will complain about the "+1" spam, and feel free to open a new issue about this. Complaints about +1 spams are at least easy to disregard. When you open a discussion about the complaints, it is not disregarded until some thought has been given to it. You are pouring oil on the fire.

**Comment by bobber205** [comment]
+1

**Comment by altarrok** [comment]
> > > To people commenting "+1" or asking for updates repeatedly: this is accomplishing nothing, except annoying both the maintainers and the people who subscribed to this issue in order to get actual updates in due time. Just stop.
> > 
> > 
> > by initiating a discussion unrelated to this issue, on this issue, you are contributing to the spam. As 5+ comments have been made about this discussion. Please don't respond and accept that +1 ers will +1, and feel free to open a new issue about this. +1 spams are at least easy to disregard, when you open a discussion it is not disregarded until some thought has been given to it. You are pouring oil to the flame..
> 
> by initiating a discussion unrelated to this issue, on this issue, you are contributing to the spam. As 6+ comments have been made about this discussion. Please don't respond and accept that people will complain about the "+1" spam, and feel free to open a new issue about this. Complaints about +1 spams are at least easy to disregard. When you open a discussion about the complaints, it is not disregarded until some thought has been given to it. You are pouring oil on the fire.

>Complaints about +1 spams are at least easy to disregard. 

**no, they aren't.** that's where your "copy opponent's move" type of thinking fails..
Also, a person that thinks someone will stop +1 ing through their comment, would be more open to stop spamming, by reading another comment about why they are wrong. So my criticism is positive, yours is negative.

**Comment by mateoghione99** [comment]
We are tired of waiting 5 years for this issue to be resolved. Everyone on planet earth should write +1

**Comment by bobber205** [comment]
A global pandemic came and went w/o a basic feature of this -- I know some PRs have been submitted I think at least to address this but havent' been merged -- what is the official stance of the team on on this? It's not like it's been Abandoned far from it.

**Comment by LeulAria** [comment]
God this issue still exist hoe can i unsubscribe 🤣🤣🤣🤣 use kysely guys just use kysely

**Comment by bobber205** [comment]
> God this issue still exist how can i unsubscribe 🤣🤣🤣🤣 use kysely guys just use kysely

<img width="346" alt="Image" src="https://github.com/user-attachments/assets/85b1b7ac-d827-42ad-bdb3-4ed38efb01ca" />

**Comment by SarenT** [comment]
> > > > To people commenting "+1" or asking for updates repeatedly: this is accomplishing nothing, except annoying both the maintainers and the people who subscribed to this issue in order to get actual updates in due time. Just stop.
> > > 
> > > 
> > > by initiating a discussion unrelated to this issue, on this issue, you are contributing to the spam. As 5+ comments have been made about this discussion. Please don't respond and accept that +1 ers will +1, and feel free to open a new issue about this. +1 spams are at least easy to disregard, when you open a discussion it is not disregarded until some thought has been given to it. You are pouring oil to the flame..
> > 
> > 
> > by initiating a discussion unrelated to this issue, on this issue, you are contributing to the spam. As 6+ comments have been made about this discussion. Please don't respond and accept that people will complain about the "+1" spam, and feel free to open a new issue about this. Complaints about +1 spams are at least easy to disregard. When you open a discussion about the complaints, it is not disregarded until some thought has been given to it. You are pouring oil on the fire.
> 
> > Complaints about +1 spams are at least easy to disregard.
> 
> **no, they aren't.** that's where your "copy opponent's move" type of thinking fails.. Also, a person that thinks someone will stop +1 ing through their comment, would be more open to stop spamming, by reading another comment about why they are wrong. So my criticism is positive, yours is negative.

Your comment is not clearly written and I won't bother to put too much effort to understand it to prove a point.

**Comment by SarenT** [comment]
-1

**Comment by mateoghione99** [comment]
The truth is that the person who made the post has retired from work. Let’s hope that their great-grandchildren, at the very least, will be able to use Prisma with geolocation.

**Comment by Ehbraheem** [comment]
Another missing feature in Prisma 😀

**Comment by sovetski** [comment]
Any update 5 years later please?...

**Comment by persiklover** [comment]
+1

**Comment by heyyfurqan** [comment]
Still missing this feature in prisma :)

**Comment by emwadde** [comment]
+1

**Comment by vladislavbogomolov** [comment]
+1

**Comment by QuantumGlitch** [comment]
+1

**Comment by garretteklof** [comment]
folks.. I hate to play pallbearer to this tragic news.. but, I've been on _this_ thread and other [threads](https://github.com/prisma/prisma/issues/2789), [just like this](https://github.com/prisma/prisma/issues/18281) for what _feels_ (and may, quite possibly be) millennias (and before most of you were born, to boot, mind you!)

I'm sorry; this is simply: **not** happening. We must, accept.

And in that may we, finally... ✨ heal ✨

**Comment by acidjazz** [comment]
Nonsense! one day my grandkids will use prisma Geo/Spatial support even though i switch to drizzle a LONG time ago.

**Comment by PetyaOgurkin** [comment]
+1

**Comment by developerDoor** [comment]
+1

**Comment by SalihCanBinboga** [comment]
+1

**Comment by ChoiWheatley** [comment]
I'm dying to wait for this index coming out

**Comment by ofekBerko** [comment]
+1

**Comment by jRexhmati** [comment]
Please add support for this!
We've been waiting on this for a long time now.

**Comment by amyewang** [comment]
+1!

**Comment by LeonardoDB** [comment]
+1

**Comment by marcospgp** [comment]
is there an escape hatch for this? as in, can I have columns with geo types in my schema and have the migration tooling not break?

**Comment by Riki4321** [comment]
+1

**Comment by ghost** [comment]
Here is a work around for get / create / update (for SQL Server) - 

```
static async get(id) {
    const record = await prisma.collection.findUnique({
        where: { ID: parseInt(id) },
        select: DETAIL_COLUMNS
    });

    // If you need geolocation data, fetch it separately with raw SQL
    if (record {
        const geoData = await prisma.$queryRaw`
            SELECT Geolocation.STAsText() as GeolocationText 
            FROM collection 
            WHERE ID = ${parseInt(id)}
        `;
        
        record.geo = geoData.GeolocationText;    
    }

    return record;
}

static async create(data) {
       // regular create / update function

      const { longitude, latitude } = data;
      if(longitude && latitude) {
          await prisma.$executeRaw`
            UPDATE collection 
            SET Geolocation = geography::STPointFromText('POINT(' + CAST(${longitude} AS VARCHAR(50)) + ' ' +   CAST(${latitude} AS VARCHAR(50)) + ')', 4326)
            WHERE ID = ${data.ID}`;
      }

      // return whatever
}
```

Hope this helps

**Comment by goktug7913** [comment]
We're adding telematics stuff, and the lack of geojson support in prisma has become a major issue for us, it's a bit surprising that this isn't given priority. Ingesting coords from 1000+ vehicles and processing data in realtime with a sliding window is kind of painful right now x(

**Comment by sss-Mihail-sss** [comment]
Have the developers given up on this issue?

**Comment by Toulix** [comment]
Any update as of 2026?

**Comment by nikp29** [comment]
+1

**Comment by lh0x00** [comment]
I’ve created a PR for this feature. I mentioned it before, but I only just found the time to work on it today. If you’re happy with it, feel free to leave a reaction and follow me—really appreciate it!
PRs: https://github.com/prisma/prisma/pull/29365 https://github.com/prisma/prisma-engines/pull/5797 https://github.com/prisma/web/pull/7685

**Comment by khalidsaidi** [comment]
A2ABench has an accepted answer for this imported thread.
- Thread: https://a2abench-api.web.app/q/cmmlg248p00j0uexhuzxxo2pv
- Accepted at: 2026-03-25T19:15:20.424Z
- Accepted answer agent: `ai village`
- Answer preview: "No matching threads were found for this query."

**Comment by maroshvarchola** [comment]
Wtf is so big problem to implement this? I mean, do we really need to switch from prisma? Like really? Why? Sorry but this is so unprofessional...

**Comment by lh0x00** [comment]
I created PR this features and using fork version to waiting for merge :)

**Comment by acidjazz** [comment]
> Wtf is so big problem to implement this? I mean, do we really need to switch from prisma? Like really? Why? Sorry but this is so unprofessional...

drizzle has been great!

**Comment by lky141319** [comment]
Claiming bounty. 
Wallet: 0xd2268DEDca16b08aF4bb6Ad20cBa877D10cf8Ac6
GitHub: @lky141319
